### PR TITLE
Include webapp build in maven build

### PR DESCRIPTION
**Pull request by mcataford**
## Description
- Cleans up `pom.xml`
- Sets up the `npm` build calls inside the Maven build process to be able to build everything only using `mvn install`.
- Cleans up `Makefile` to remove redundant recipe `build_all` since `build` is all that is needed to build the plugin now.
- Update CI to reflect `build_all` deprecation (uses `build`).

### DevQA Prep
Set up a clean environment using the `maven:3.3-jdk-8` Docker image. Install `git` and pull down the repo. This way, you can ensure that the build isn't affected by globally-installed packages or any kind. Additionally, this replicates what happens in CI.

### DevQA Steps
- Run `mvn install` or `mvn clean install`;
- Note that `npm` is installed during the build process and that the webapp is built in the `compile` phase.
- Additionally, you can test that the `hpi` artifact is viable (although observing that `src/main/webapp` is well-formed and that the build tests pass is sufficient to affirm confidently that the `hpi` is viable).

### Comments:
- The Maven build might be optimization a bit more, as its plugin system sometimes introduces reruns or certain `execution` blocks. This can be addresses later on.
- Another issue that was addressed was `maven-surefire` using the system classloader and failing on tests.
- The `make build` was also bypassed entirely.

### CODE REVIEW

**Comment by mcataford** [comment]
As of https://github.com/tophat/jenkins-timeline-plugin/pull/58/commits/dc2e704f0491255ef2f0b0c6e46383c02396c3e3, building using `mvn install` has the entire build process working fine, current issue is stage duplication.

TODO:
- Ensure that the `npm` command to build to plugin force copies to overwrite, or cleans the target before copying.

**Comment by mcataford** [comment]
Ready for self-QA and CR/QA as of https://github.com/tophat/jenkins-timeline-plugin/pull/58/commits/a61c7ad5c6684d779c27197bc47c5686fb5d6c87. `hpi` seems viable.

**Comment by mcataford** [comment]
Passes self-QA, adjusted version number.

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,14 +43,15 @@ jobs:
   test:
     working_directory: ~/jenkins-timeline-plugin
     docker:
-      - image: maven:3.6.0-jdk-8-alpine
+      - image: maven:3.6.0-jdk-8
```
Comment (jakebolam) [inline .circleci/config.yml]: So much trouble with alpine images. We've stopped using them in production too

**Pull request merged**